### ISSUE TITLE
allow use of entering host/ip directly when detection fails

### ISSUE THREAD

**Issue by maxandersen** [elgatosf/streamdeck-philipshue#3]
On my network philips hue bridges historically had a hard time being discovered reliably. Apps that allow for entering ip/host directly have worked much better. Would be great if the elgato plugin would allow for that too.

**Comment by Kakifrucht** [comment]
Related to #2, doesn't seem like Elgato cares about adding basic features like this.

**Comment by mmaechtel** [comment]
Same problem, when bridge is in different vlan! Any chance to config the ip into the settings anywhere, so stream deck finds the hue bridge again?

**Comment by Sumidu** [comment]
You can easily hack it in yourself:

1. Find the pluginfolder of the hue-plugin. On mac it is `~/Library/Application Support/com.elgato.StreamDeck/Plugins/com.elgato.philipps-hue.sdPlugin`.

2. Find the IP address and ID from the Hue App on your phone.

3. Replace the "Bridge.discover" function in the `plugin/js/philips/meethue.js` file with the following code.

```javascript
Bridge.discover = function(callback) {
    // this is where my code just always detects a bridge
    var bridges = [];
    bridges.push(new Bridge("192.yourip.add.ress", "00178yourid"));
    callback(true, bridges);
   // everything else here removed.. :)
  
};
```

This is how I fixed it. Actually I copied the whole plugin and renamed things everywhere, so I can have both plugins. But if you only need to find one bridge with a fixed IP address, this helps.

**Comment by dopeghoti** [comment]
Sounds like, since this is an OS plugin, a pull request could be made to allow for this without diving into the codebase by hand.

**Comment by boydfields** [comment]
@Sumidu could you supply a copy of your meethue.js file?  I can't seem to get your mods to work. Many thanks

**Comment by Sumidu** [comment]
I no longer use this hack :(

But it was really just replacing the existing `Bridge.discover` function and adding your ip-address and your id.

It's not really sensible to make a pull request out of this hack, as it would break the existing code. My javascript capabilities are not good enough to create the interface for setting up IP and ID.. 👎

**Comment by boydfields** [comment]
Could I ask why you no longer use this hack? Didn't it stop working for you?
Thanks

**Comment by Sumidu** [comment]
No longer needed it. In my new setup I can find my Hue bridge :)

**Comment by sli** [comment]
I've implemented a rudimentary version of this in a fork. It currently only allows manual setup of a single bridge and needs a bit of visual cleanup but it's working well for me. The translations will also need to be done, I only speak English. Once it's finished I'll file a PR, but for now it can be manually built (or just drop the new setup source and translations into your current install).

There are also some translation updates in the currently released version that are not in the master branch, I've included those.

https://github.com/sli/streamdeck-philipshue
